### PR TITLE
[Feature] support save params to S3

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
Add utils.save_parameters API with s3 uri support. The feature assumes boto3 profile is configured locally, and uses the default file upload API. 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/gluonnlp/utils/parameter.py [hunk 1]
```diff
@@ -98,3 +101,52 @@ def _norm(array):
...
+    ----------
+    model : mx.gluon.Block
+        The model to save.
+    uri : str
+        Path to file.
+    """
+    S3_PREFIX = 's3://'
+    if S3_PREFIX in filename:
+        try:
+            import boto3
```
Comment (szha) [inline src/gluonnlp/utils/parameter.py]: add this to the extras in setup.py

**Comment by mli** [comment]
Job PR-700/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-700/1/index.html

**Comment by mli** [comment]
Job PR-700/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-700/2/index.html

**Comment by mli** [comment]
Job PR-700/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-700/3/index.html

**Comment by mli** [comment]
Job PR-700/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-700/4/index.html

**Comment by mli** [comment]
Job PR-700/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-700/5/index.html

**Comment by mli** [comment]
Job PR-700/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-700/6/index.html

**Comment by mli** [comment]
Job PR-700/7 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-700/7/index.html

**Comment by mli** [comment]
Job PR-700/14 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-700/14/index.html

**Pull request merged**